src/main/java/bio/terra/stairway/Stairway.java

```diff
@@ -181,12 +182,18 @@ public Builder flightFactory(FlightFactory flightFactory) {
     }
 
     /**
+     * Each call to stairwayHook adds hook object to a list of hooks. The hooks are processed in the
+     * order in which they are added to the builder.
+     *
      * @param stairwayHook object containing hooks for logging at beginning and end of flight and
      *     step of stairway
      * @return this
      */
     public Builder stairwayHook(StairwayHook stairwayHook) {
-      this.stairwayHook = stairwayHook;
+      if (this.stairwayHooks == null) {
+        this.stairwayHooks = new ArrayList<>();
```

Comment:
Yes. I'm doing premature optimization by avoiding a tidbit of code in the case where no hooks are defined. Since it looks like we will always have at least trace hooks defined in production, that is entirely useless.